wheel/python/clvm_rs/tree_hash.py

```diff
@@ -0,0 +1,107 @@
+"""
+This is an implementation of `sha256_treehash`, used to calculate
+puzzle hashes in clvm.
+
+This implementation goes to great pains to be non-recursive so we don't
+have to worry about blowing out the python stack.
+"""
+
+from hashlib import sha256
+from typing import Callable, List, Tuple, cast
+
+from .clvm_storage import CLVMStorage
+
+
+OP_STACK_F = Callable[[List[CLVMStorage], List[bytes], List["OP_STACK_F"]], None]
+
+
+class Treehasher:
```

Comment:
it seems like a bit of a detour to create a class to record the tree-hash format, just to instantiate a singleton and pull out the member functions. Why not just write the free functions directly?
You don't reuse this class anywhere, and we only have one form of tree-hash (and I think we would need to have a really good reason to add another kind, with values other than `1` and `2`)

Reply:
See above. It's motivated by minimizing the number of things `Program` needs to use, and tie all those magic values into one class (not too different from a module except an instance is instantiated).